lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1183,44 +1185,41 @@ private void onChannelMessage(ProtocolMessage message) {
-         * this will be reattached in setConnection() */
-        ErrorInfo error = message.error;
-        if(connection.id != null && !message.connectionId.equals(connection.id)) {
-            /* we need to suspend the original connection */
-            if(error == null) {
-                error = REASON_SUSPENDED;
+        final ErrorInfo error = message.error;
+        connection.reason = error;
+        if (connection.id != null) { // there was a previous connection, so this is a resume and RTN15c applies
+            Log.d(TAG, "There was a connection resume");
+            if(message.connectionId.equals(connection.id)) {
+                // resume succeeded
+                if(message.error == null) {
+                    // RTN15c1: no action required wrt channel state
+                    Log.d(TAG, "connection has reconnected and resumed successfully");
+                } else {
+                    // RTN15c2: no action required wrt channel state
+                    Log.d(TAG, "connection resume success with non-fatal error: " + error.message);
+                }
+                // Add pending messages to the front of queued messages to be sent later
+                addPendingMessagesToQueuedMessages(false);
+            } else {
+                // RTN15c3: resume failed
+                if (error != null){
+                    Log.d(TAG, "connection resume failed with error: " + error.message);
+                }else { // This shouldn't happen but, putting it here for safety
+                    Log.d(TAG, "connection resume failed without error" );
+                }
+
+                channels.reattachOnResumeFailure();
+                // Add any messages still pending from the previous transport (RTN19a) to the front of queued messages
+                // however, this time the pending messages have to have newly assigned `
+                // msgSerial`s. They can't simply be replayed, as they are in the successful resume case
+                addPendingMessagesToQueuedMessages(true);
             }
-            channels.suspendAll(error, false);
         }
 
-        /* set the new connection id */
-        ConnectionDetails connectionDetails = message.connectionDetails;
-        connection.key = connectionDetails.connectionKey;
-        if (!message.connectionId.equals(connection.id)) {
-            /* The connection id has changed. Reset the message serial and the
-             * pending message queue (which fails the messages currently in
-             * there). */
-            pendingMessages.reset(msgSerial,
-                    new ErrorInfo("Connection resume failed", 500, 50000));
-            msgSerial = 0;
-        }
         connection.id = message.connectionId;
-        if(message.connectionSerial != null) {
-            connection.serial = message.connectionSerial.longValue();
-            if (connection.key != null)
-                connection.recoveryKey = connection.key + ":" + message.connectionSerial;
```

Comment:
Why was this removed? Where else is `recoveryKey` set?

Reply:
This was removed by mistake - as part of changes from protocol-2.0. Introduced it back bc2cfef966abec1abaf02d8037e637fd51527484